### PR TITLE
Wait for the CMS to be ready before triggering govdelivery

### PR DESCRIPTION
**Pull request by cweagans**
## Description

In the past few days, there have been a number of content release failures related to trying to do stuff while the CMS is deploying.

Here are a couple of examples:

* https://github.com/department-of-veterans-affairs/content-build/actions/runs/1555528870
* https://github.com/department-of-veterans-affairs/content-build/actions/runs/1532069013

Instead of just sending the request, let's wait for the CMS to be ready and _then_ send the request. This needs some testing, which I will do in a separate PR so as not to pollute the Git history.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by cweagans** [comment]
https://github.com/department-of-veterans-affairs/content-build/runs/4463284377?check_suite_focus=true correctly timed out after 30 seconds.

I just pushed 057bf44 and I'm now waiting for https://github.com/department-of-veterans-affairs/content-build/runs/4463306325?check_suite_focus=true, which should time out after 30 _minutes_.

**Comment by cweagans** [comment]
https://github.com/department-of-veterans-affairs/content-build/runs/4463306325?check_suite_focus=true correctly timed out after 30 minutes.

https://github.com/department-of-veterans-affairs/content-build/runs/4463724475?check_suite_focus=true proceeded immediately with a working URL.

File: .github/workflows/wait-for-cms-ready/action.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: Wait for CMS to be ready
+description: Pause until we know that the CMS is online and ready to go
+
+inputs:
+  base_url:
+    description: base url for the cms environment to wait for
+    required: false
+    default: 'https://prod.cms.va.gov'
+  interval:
+    description: how often to poll the environment for status (in seconds)
```
Comment (oseasmoran73) [inline .github/workflows/wait-for-cms-ready/action.yml:10]:
```suggestion
    description: How often to poll the environment for status (in seconds)
```

File: .github/workflows/wait-for-cms-ready/action.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    required: false
+    default: '15'
+  timeout:
+    description: how long should we wait for the cms to be ready (in seconds)
```
Comment (oseasmoran73) [inline .github/workflows/wait-for-cms-ready/action.yml:14]:
```suggestion
    description: How long should we wait for the cms to be ready (in seconds)
```

File: .github/workflows/wait-for-cms-ready/action.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Wait for CMS to be ready
+description: Pause until we know that the CMS is online and ready to go
+
+inputs:
+  base_url:
+    description: base url for the cms environment to wait for
```
Comment (oseasmoran73) [inline .github/workflows/wait-for-cms-ready/action.yml:6]:
```suggestion
    description: Base url for the cms environment to wait for
```

**Review by oseasmoran73 — COMMENTED** [review]
I think with this composite action created, we should move the callback function into it and rename the flow to be `GovDelivery Callback` or rather than it just being `Wait for CMS to be ready` thoughts?

**Comment by cweagans** [comment]
I thought about that. I didn't go back too far in the logs to find other failed builds but I wasn't sure if we might need this action before the graphql requests too. It's entirely possible that if a runner takes a bit to start up, the graphql requests wouldn't be done before the cms went down for deployment.

Can we try this and move govdelivery later if we don't run into that problem in the next couple weeks?

**Comment by oseasmoran73** [comment]
@cweagans Sure, can you add a `TODO` comment and write track somewhere in a ticket?

**Comment by cweagans** [comment]
Ticket is here: https://github.com/department-of-veterans-affairs/va.gov-team/issues/33873

Todo comment commit is here:  a774080

**Review by oseasmoran73 — APPROVED** [review]

**Pull request merged**